salesagility/SuiteCRM
# Add dotenv support for the test environment

## Description
Fixes #7620. This introduces support for loading environment variables via [phpdotenv](https://github.com/vlucas/phpdotenv).

I chose phpdotenv over Symfony dotenv because Symfony Dotenv doesn't support overloading existing environment variables.

## Motivation and Context

If you wanted to have multiple CRMs locally, or a separate configuration when running the test suite vs. developing the CRM, that wasn't possible without manually updating your `~/.bash_profile` or updating your `config.php` file.

That makes running the tests or contributing upstream unnecessarily difficult. With a dotenv file, we can fix both these problems :)

Simply copy the `.env.dist` file to `.env.test` in the root of your project and modify each variable as necessary to set up your project. The environment variables defined in `.env.test` only apply to the test suites, and will override any environment variables set elsewhere (e.g. in `~/.bash_profile`). If the `.env.test` file doesn't exist, the CRM works the same as before.

## How To Test This
- Copy the `.env.dist` file to `.env.test`, modify it to fit your existing environment if necessary.
- Run the tests (e.g. `./vendor/bin/robo test:acceptance`).

You can modify the `tests/_bootstrap.php` file to include something like `echo getenv('INSTANCE_URL');` after the dotenv file is loaded, and then see if the tests actually output the instance URL set in your `.env.test` file. If they do, it worked :D

You can also make sure the tests continue to work if the `.env.test` doesn't exist.

I'm not 100% sure whether these environment variables are loaded with PHPUnit tests. I need to look into that a bit more.

## Types of changes

### Final checklist

I should probably update the automated testing docs to mention this improvement.
